src/SoftRobots.Inverse/component/constraint/BarycentricCenterEffector.inl

```diff
@@ -77,31 +76,27 @@ BarycentricCenterEffector<DataTypes>::~BarycentricCenterEffector()
 template<class DataTypes>
 void BarycentricCenterEffector<DataTypes>::init()
 {
-    Inherit1::init();
-
-    if(m_state==nullptr)
-        msg_error() << "There is no mechanical state associated with this node. "
-                       "the object is deactivated. "
-                       "To remove this error message fix your scene possibly by "
-                       "adding a MechanicalObject." ;
-    initData();
-
-    d_drawBarycenter.setGroup("Visualization");
-    d_barycenter.setReadOnly(true);
+    // d_indices is inherited from PositionEffector.
+    // Even though we do not use this data we need to initilize it to avoid warnings
+    auto indices = sofa::helper::getWriteAccessor(d_indices);
+    indices.resize(1);
 
-    computeBarycenter();
-}
+    Inherit1::init();
 
-template<class DataTypes>
-void BarycentricCenterEffector<DataTypes>::reinit()
-{
-    initData();
+    if (d_axis.isSet())
+    {
+        msg_deprecated() << "The data axis is deprecated. To fix your scene please use useDirections instead. It will be remove in v25.06.";
```

Comment:
We recently introduced `RenamedData`, but I am not convinced that it works well...

I wonder if this check is to be done in the parse method...